File: src/main/java/me/taubsie/dungeonhub/application/listener/BotReloadListener.java
package me.taubsie.dungeonhub.application.listener;

import me.taubsie.dungeonhub.application.connection.DiscordConnection;
import org.javacord.api.event.connection.ReconnectEvent;
import org.javacord.api.event.connection.ResumeEvent;
import org.javacord.api.listener.connection.ReconnectListener;
import org.javacord.api.listener.connection.ResumeListener;

@Listener
public class BotReloadListener implements ReconnectListener, ResumeListener {
    @Override
    public void onReconnect(ReconnectEvent event) {
        DiscordConnection.getInstance().resetBotAppearance();
    }

    @Override
    public void onResume(ResumeEvent event) {
        DiscordConnection.getInstance().resetBotAppearance();
    }
}

File: src/main/java/me/taubsie/dungeonhub/kord/application/listener/BotReloadListener.kt
package me.taubsie.dungeonhub.kord.application.listener

import com.kotlindiscord.kord.extensions.extensions.Extension
import com.kotlindiscord.kord.extensions.extensions.event
import dev.kord.core.event.gateway.ResumedEvent
import me.taubsie.dungeonhub.kord.application.connection.DiscordConnection
import me.taubsie.dungeonhub.kord.application.loader.LoadExtension

@LoadExtension
class BotReloadListener : Extension() {
    override val name = "bot-reload-listener"

    override suspend fun setup() {
        event<ResumedEvent> {
            action {
                DiscordConnection.resetBotAppearance()
            }
        }
    }
}